src/database/index.tsx

```diff
@@ -123,6 +119,68 @@ const buildUpdatePayload = (entity: any) => {
   return payload;
 };
 
+const timestampToDate = (timestamp: firebase.firestore.Timestamp): Date => {
+  return timestamp.toDate();
+};
+
+const buildPlannedRelease = (plannedReleaseData: any): PlannedRelease => {
+  let plannedRelease: PlannedRelease = plannedReleaseData;
+
+  const plannedReleaseDataDate = plannedReleaseData.date;
+  if (plannedReleaseDataDate instanceof firebase.firestore.Timestamp) {
+    plannedRelease.date = timestampToDate(plannedReleaseDataDate);
+  } else {
```

Comment:
So this else assumes that the only other type option is a string, right? Is that a safe assumption on the way out of Firestore?

Reply:
It should be a safe assumption because that's the only thing we previously stored for the Planned Release dates.  That said, I can turn this into an `else if` to check for String explicitly and throw an exception in the final `else` case.

Reply:
I think I would like that defensiveness--in either approach we get an error but in this approach we can better differentiate between the two error cases of "not an expected type" and "not a parseable string."